### PR TITLE
promises for upload

### PR DESCRIPTION
**Pull request by alexjohnson505**

### CODE REVIEW

File: api/entries.js [hunk 1]
```diff
@@ -40,7 +41,7 @@ var saveEntry = function(data) {
                     if (err) {
                         console.log(err);
                     } else {
-                        return console.log('saved');
+                        deferred.resolve(console.log('saved'));
```
Comment (alexjohnson505) [inline api/entries.js:44]: Do we wanna resolve the promise on error as well?

File: api/entries.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ var Q = require('q');
 
 // save entry helper
```
Comment (alexjohnson505) [inline api/entries.js:5]: Needs better function description. Ex: Creates promise used for tracking multi-step .... etc

File: api/entries.js [hunk 3]
```diff
@@ -60,6 +61,7 @@ module.exports = {
 
     multiPost: function(req, res) {
 
+        var promises = [];
```
Comment (alexjohnson505) [inline api/entries.js:64]: Gnarly, but I love it.

**Comment by pragsanshi** [comment]
+1

**Pull request closed without merging**